cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_train_dense_default_impl.i

```diff
@@ -1029,10 +1696,10 @@ public:
 //////////////////////////////////////////////////////////////////////////////////////////
 // ClassificationTrainBatchKernel
 //////////////////////////////////////////////////////////////////////////////////////////
-template <typename algorithmFPType, Method method, CpuType cpu>
-services::Status ClassificationTrainBatchKernel<algorithmFPType, method, cpu>::compute(
-    HostAppIface * pHostApp, const NumericTable * x, const NumericTable * y, const NumericTable * w, decision_forest::classification::Model & m,
-    Result & res, const decision_forest::classification::training::Parameter & par)
+template <typename algorithmFPType, Method method, CpuType cpu, typename helper>
+services::Status _compute(HostAppIface * pHostApp, const NumericTable * x, const NumericTable * y, const NumericTable * w,
+                          decision_forest::classification::Model & m, Result & res, const decision_forest::classification::training::Parameter & par,
+                          bool memSave)
```

Comment:
It seems there is no need in defining a new `_compute` method, it is possible just to add a `memSave` boolean variable into the original implementation and define it properly depending on the splitter  type.

Reply:
Agreed. Because of the now divergent nature between best and random, its best to keep all the logic in 1 function.